lib/socket.ts

```diff
@@ -95,34 +111,38 @@ export function createSocket(options: ConnectionOptions): Promise<HaWebSocket> {
     };
 
     const handleMessage = async (event: MessageEvent) => {
-      const message = JSON.parse(event.data);
-
-      if (DEBUG) {
-        console.log("[Auth phase] Received", message);
+      let message_group: WebSocketAuthMessage | [WebSocketAuthMessage] = JSON.parse(event.data);
```

Comment:
Let's not support message coalescing during auth. I think that it's also a different piece of code in `core` anyway than the one you touched.